lib/beaker/subcommand.rb

```diff
@@ -6,15 +6,41 @@ module Beaker
   class Subcommand < Thor
     SubcommandUtil = Beaker::Subcommands::SubcommandUtil
 
+    def initialize(*args)
+      super
+      @@config = SubcommandUtil.load_config()
+    end
+
     desc "init", "Initialises the beaker test environment configuration"
     option :hypervisor, :type => :string, :required => true
     long_desc <<-LONGDESC
-      Initialises a beaker environment configuration
+    Initialises a beaker environment configuration
     LONGDESC
     def init()
       SubcommandUtil.verify_init_args(options)
       SubcommandUtil.require_tasks()
       SubcommandUtil.init_hypervisor(options)
+      SubcommandUtil.store_config(options)
+      SubcommandUtil.delete_config([:provisioned, :hosts])
+    end
+
+    desc "provision", "Provisions the beaker test environment"
+    option :force, :type => :boolean
```

Comment:
If the user has already provisioned their nodes, then re-running provision will exit with a message stating that the nodes have already been provisioned (see below), but I thought that there might be scenarios where the user might also want a way to re-provision nodes, so I added a --force flag